pandas/tslib.pyx

```diff
@@ -291,10 +291,25 @@ class Timestamp(_Timestamp):
     def combine(cls, date, time):
         return cls(datetime.combine(date, time))
 
-    def __new__(cls, object ts_input, object offset=None, tz=None, unit=None):
+    def __new__(cls,
+            # Keep legacy parameter names.
+            object ts_input, object offset=None, tz=None, unit=None,
+            # Switch to datetime positional paramers, starting with `minute`.
+            minute=0, second=0, microsecond=0, tzinfo=None):
```

Comment:
ok.

I would ideally I think like to have this signature:

``ts_input=None, offset=None, tz=None, unit=None, year=None, month=None, day=None, hour=None, minute=None, second=None, tzinfo=None``

then its simple to do kwargs, e.g. you MUST provide ``year`` is not None (and all of ts_input, offset, tz, unit must be ``None``).

Then if you are actually creating a ``datetime.time`` (thru either positional or kwargs), then you can just call like (after you have done a rename of args for the positional if needed).

e.g.

```datetime.datetime(year=year, month=month, day=day, hour=hour or 0, minute=minute or 0, second=second or 0)``